testng-eclipse-plugin/src/main/org/testng/eclipse/ui/TestRunnerViewPart.java

```diff
@@ -372,23 +378,53 @@ public void run() {
               });
             }
             catch(SocketTimeoutException ex) {
+              TestNGPlugin.log(ex);
               postSyncRunnable(new Runnable() {
                 public void run() {
                   boolean useProjectJar =
                       TestNGPlugin.getPluginPreferenceStore().getUseProjectJar(project.getProject().getName());
                   String suggestion = useProjectJar
-                     ? "Uncheck the 'Use Project testng.jar' option from your Project properties and try again."
-                     : "Make sure you don't have an older version of testng.jar on your class path.";
-                  new ErrorDialog(m_counterComposite.getShell(), "Couldn't launch TestNG",
-                      "Couldn't contact the RemoteTestNG client. " + suggestion,
-                      new StatusInfo(IStatus.ERROR, "Timeout while trying to contact RemoteTestNG."),
-                      IStatus.ERROR).open();
+                      ? ResourceUtil.getString("TestRunnerViewPart.message.suggestion1")
+                      : ResourceUtil.getString("TestRunnerViewPart.message.suggestion2");
+                   new ErrorDialog(m_counterComposite.getShell(),
+                       "Couldn't launch TestNG", suggestion,
+                       new Status(IStatus.ERROR, TestNGPlugin.PLUGIN_ID,
+                           ResourceUtil.getString("TestRunnerViewPart.message.reason")),
+                       IStatus.ERROR).open();
                 }
               });
             }
             return Status.OK_STATUS;
           }
+
+          @Override
+          protected void canceling() {
+            if (initialized == false) {
+              TestNGPlugin.log("stop the socket server as it's still pending on accpeting the client");
+              messageMarshaller.stopReceiver();
+            }
+          }
+        };
+
+        // handle the process terminated event
+        IDebugEventSetListener listener = new IDebugEventSetListener() {
+          public void handleDebugEvents(DebugEvent[] events) {
+            for (DebugEvent event : events) {
+              if (event.getKind() == DebugEvent.TERMINATE) {
+                Object source = event.getSource();
+                if (source instanceof IProcess) {
+                  IProcess process = (IProcess) source;
+                  if (process.getLaunch().equals(launch)) {
+                    testRunListeningJob.cancel();

```

Comment:
hook into eclipse launch lifecycle, cancel the job if the launch terminated (e.g. the process end normally, or people killed it)